LazyAlchemyLearner.lua

```diff
@@ -142,41 +187,164 @@ local function getItemLinkFromItemId(itemId)
+		if((bag + bank + craft) > 0) then
+			return bag + bank + craft, solvents[i], i
+		end
 
-	for i = start, 1 , -1 do
-		local bag, bank, craft = GetItemLinkStacks(getItemLinkFromItemId(poison[i]))
-		return bag + bank + craft, poison[i],  i
+		-- potions
+		local bag, bank, craft = GetItemLinkStacks(getItemLinkFromItemId(solvents[i+9]))
+		if((bag + bank + craft) > 0) then
+			return bag + bank + craft, solvents[i+9], i
+		end
 	end
-	return nil, nil,nil
+	
+	return nil, nil, nil
+end
+
+local function GetAvailableReagent(reagentId)
+	local bag, bank, craft = GetItemLinkStacks(getItemLinkFromItemId(reagentId))
+	if((bag + bank + craft) > 0) then
+		return bag + bank + craft
+	end
+	return nil
+end
+
+-- Function to find matching traits between two reagents
+local function GetMatchingTraits(reagent1, reagent2)
+    local traits1 = reagentTraits[reagent1]
+    local traits2 = reagentTraits[reagent2]
+    local matchingTraits = {}
+
+    -- Compare the traits from both reagents
+    for _, trait1 in ipairs(traits1) do
+        for _, trait2 in ipairs(traits2) do
+            if trait1 == trait2 then
+                table.insert(matchingTraits, trait1)
+            end
+        end
+    end
+    return matchingTraits
+end
+
+-- Function to check if a value exists in a table
+local function Contains(table, value)
+    for _, v in ipairs(table) do
+        if v == value then
+            return true
+        end
+    end
+    return false
 end
 
 local function alchemyQueuer(combos)
 	local LLC = LazyAlchemyLearner.LLC
 	local remainingSolvent = 0
 	local solvent
-	local position= math.min(GetNonCombatBonus(NON_COMBAT_BONUS_ALCHEMY_LEVEL) + 1, #poison) + 1
+	local position = 1
 	local queued = 0
+	
+	--I am adding a solvent check here to see if the player has any solvent at all, no need to process anything if nothing will be queued anyway
+	remainingSolvent, solvent, position = getSolvent(GetNonCombatBonus(NON_COMBAT_BONUS_ALCHEMY_LEVEL), position)
+	
+	if(remainingSolvent == nil) then
+		d("You need solvents to craft potions")
+		return queued
+	end
+	
 	for i = 1, #combos do
 		local known = true
+		-- we check what traits between the two reagents are matching
+		local theoreticalMatchingTraits = GetMatchingTraits(combos[i][1], combos[i][2])
+		local amountMatchingReagant1 = 0
+		local amountMatchingReagant2 = 0
 		for j = 1, 4 do
-			known = known and GetItemLinkReagentTraitInfo(getItemLinkFromItemId(comboInfo[i][1]), j) and 
-			GetItemLinkReagentTraitInfo(getItemLinkFromItemId(comboInfo[i][2]), j)
-		end
-		if remainingSolvent == 0 then
-			local solventProficiency = GetNonCombatBonus(NON_COMBAT_BONUS_ALCHEMY_LEVEL)
-			if solventProficiency == 0 then
-				solventProficiency = 1
+		
+			--[[
+				k = known status of the reagant's trait {true/false}
+				n = name of the trait, currently the name wil be nil if the trait is unknown
+			--]]
+			local k1, n1 = GetItemLinkReagentTraitInfo(getItemLinkFromItemId(combos[i][1]), j)
```

Comment:
You're using it now, so it's not relevant anymore, but in the future if there's a return you don't care about, you can do
local _, n1 = functionCall()

Reply:
Ah thanks for the tip, I was wondering about that